### PR TITLE
BFT chain unit tests reconfigure

### PR DESCRIPTION
**Pull request by MayRosenbaum**
#### Type of change

- Test update

#### Description

This PR aims to test reconfiguration cases: 
1. Add a new node to the network 
2. Configure the network while a node is down, in such case the node has to sync with other nodes in the network about the config change.

#### Related issues

issue #4008

### CODE REVIEW

File: orderer/consensus/smartbft/chain.go [hunk 1]
```diff
@@ -167,6 +167,17 @@ func NewChain(
 		return nil, errors.Wrap(err, "failed to verify SmartBFT-Go configuration")
 	}
 
+	c.clusterService = &cluster.ClusterService{
```
Comment (yacovm) [inline orderer/consensus/smartbft/chain.go]: why are you re-defining this?

Comment (MayRosenbaum) [reply orderer/consensus/smartbft/chain.go]: Will be updated in a "post-review" commit as we discussed.

File: orderer/consensus/smartbft/util_network_test.go [hunk 2]
```diff
@@ -516,34 +616,52 @@ func createConfigBlock(t *testing.T, channelId string) *cb.Block {
 	require.NoError(t, err)
 	require.NotNil(t, channelGroup)
 
+	// update organization endpoints
+	ordererEndpoints := channelGroup.Groups[channelconfig.OrdererGroupKey].Groups["SampleOrg"].Values["Endpoints"].Value
+	ordererEndpointsVal := &cb.OrdererAddresses{}
+	proto.Unmarshal(ordererEndpoints, ordererEndpointsVal)
```
Comment (tock-ibm) [inline orderer/consensus/smartbft/util_network_test.go]: check the error out of proto.Unmarshal

File: orderer/consensus/smartbft/util_network_test.go [hunk 3]
```diff
@@ -74,6 +97,30 @@ func (ns *NetworkSetupInfo) CreateNodes(numberOfNodes int) map[uint64]*Node {
...
+	newNodeId := numberOfNodes
+
+	ns.t.Logf("Adding node %v to the network", newNodeId)
+
+	var ledgerToSyncWith []*cb.Block
+	for _, n := range ns.nodeIdToNode {
+		if n.NodeId != newNodeId {
+			ledgerToSyncWith = n.State.GetLedgerArray()
+		}
+	}
```
Comment (tock-ibm) [inline orderer/consensus/smartbft/util_network_test.go]: why are you copying over and over from every node? you have to pick one existing node with the largest ledger.

File: orderer/consensus/smartbft/util_network_test.go [hunk 4]
```diff
@@ -11,12 +11,15 @@ import (
...
 	"reflect"
+	"slices"
 	"sort"
 	"sync"
 	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/hyperledger/fabric/common/channelconfig"
+
```
Comment (tock-ibm) [inline orderer/consensus/smartbft/util_network_test.go]: remove the space

File: orderer/consensus/smartbft/util_network_test.go (hunk 3 above)
Comment (MayRosenbaum) [reply orderer/consensus/smartbft/util_network_test.go]: Right, a break command is missing here (the idea was to find some node and take its ledger), but pick the largest ledger is preferred.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**